### PR TITLE
Added e2b protocol for running agents

### PR DESCRIPTION
**Pull request by Swiftyos**
This PR introduces the E2B agent protocol, adding capabilities to start an agent, task an agent, and include a heartbeat endpoint to ensure the agent server is operational. Additionally, enhancements were made to the agbenchmark run method to accommodate these changes.

### E2B Agent Protocol:

- Implemented the mechanism to start an agent using the E2B protocol.
- Added the functionality to task an agent with specific instructions.
- Introduced a heartbeat endpoint for real-time monitoring of agent server status.
- Upgrade of agbenchmark run method:
- Modified the run method to support E2B agent tasks.

### Miscellaneous:
- Added necessary boilerplate code for E2B integration.
- added some agent template code for testing - will be removed shortly
- dropped agbenchmark down to 0.0.2 as latest version is missing the challanges

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,13 +9,15 @@ readme = "README.md"
...
 openai = "^0.27.8"
 autopack-tools = "^0.2.0"
 psutil = "^5.9.5"
-agent-protocol = "^0.2.0"
+agent-protocol = "^0.2.2"
 helicone = "^1.0.6"
-agbenchmark = "^0.0.3"
+
+tenacity = "^8.2.2"
+agent-protocol-client = "^0.2.2"
```
Comment (jakubno) [inline pyproject.toml:20]: this won't be needed

File: agbenchmark/benchmarks.py [hunk 2]
```diff
@@ -1,17 +1,36 @@
...
-from dotenv import load_dotenv
+from agent_protocol_client import Configuration
 
+import agbenchmark.e2b_boilerplate
 
-def run(task: str) -> None:
+configuration = Configuration(host="http://localhost:8915")
+
+
+async def run_specific_agent(task: str) -> None:
```
Comment (jakubno) [inline agbenchmark/benchmarks.py:11]: you can start this directly, no need to run it in subprocess. Check this https://github.com/Significant-Gravitas/Auto-GPT-Benchmarks/pull/278 and this: https://github.com/e2b-dev/smol-developer/tree/benchmarks/agbenchmark

The integration should be simple as that, but we will need new version of benchmarks

File: agbenchmark/e2b_boilerplate.py
Comment (jakubno) [inline agbenchmark/e2b_boilerplate.py:1]: This file isn't needed

File: autogpt/__main__.py [hunk 3]
```diff
@@ -1,52 +1,9 @@
...
-
-    agent_executor.run(user_input)
 
+import autogpt.agent
 
 if __name__ == "__main__":
-    main()
+    """Runs the agent server"""
+    load_dotenv()
+    autogpt.agent.start_agent(port=8915)
```
Comment (jakubno) [inline autogpt/__main__.py:8]: this is good!

File: pyproject.toml (hunk 1 above)
Comment (Swiftyos) [reply pyproject.toml:20]: At what point will this not be needed?

File: agbenchmark/benchmarks.py (hunk 2 above)
Comment (Swiftyos) [reply agbenchmark/benchmarks.py:11]: Thanks, I'll take a look at your PR on AGBenchmark see if I can get that merged asap

File: pyproject.toml (hunk 1 above)
Comment (jakubno) [reply pyproject.toml:20]: After you simplify to the API benchmark format

File: agbenchmark/benchmarks.py (hunk 2 above)
Comment (jakubno) [reply agbenchmark/benchmarks.py:11]: Thanks 🙏🏻

**Comment by Swiftyos** [comment]
Merging to keep the pace up. 
Once AgBenchmark is update I will make the changes to remove the boilerplate code.

**Pull request merged**